util/src/main/java/io/kubernetes/client/util/leaderelection/resourcelock/ConfigMapLock.java

```diff
@@ -0,0 +1,115 @@
+package io.kubernetes.client.util.leaderelection.resourcelock;
+
+import io.kubernetes.client.apis.CoreV1Api;
+import io.kubernetes.client.models.V1ConfigMap;
+import io.kubernetes.client.models.V1ObjectMeta;
+import io.kubernetes.client.util.leaderelection.LeaderElectionRecord;
+import io.kubernetes.client.util.leaderelection.Lock;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConfigMapLock implements Lock {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConfigMapLock.class);
+
+  // Namespace and name describes the configMap object
+  // that the LeaderElector will attempt to lead.
+  private final String namespace;
+  private final String name;
+  private final String identity;
+
+  private CoreV1Api coreV1Client = new CoreV1Api();
```

Comment:
Probably want to give users the option to pass this in instead of using the default client?